app/src/main/java/io/homeassistant/companion/android/sensors/CarSensorManager.kt

```diff
@@ -135,33 +192,39 @@ class CarSensorManager :
     }
 
     override fun hasSensor(context: Context): Boolean {
-        // TODO: show sensors for automotive (except odometer) once
-        //  we can ask for special automotive permissions in requiredPermissions
-        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.O &&
-            !context.packageManager.hasSystemFeature(PackageManager.FEATURE_AUTOMOTIVE) &&
-            BuildConfig.FLAVOR == "full"
+        this.context = context.applicationContext
+
+        return if (isAutomotive) {
+            BuildConfig.FLAVOR == "minimal"
+        } else {
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.O &&
+                BuildConfig.FLAVOR == "full"
+        }
     }
 
     override fun requiredPermissions(sensorId: String): Array<String> {
-        return when {
-            (sensorId == fuelLevel.id || sensorId == batteryLevel.id || sensorId == fuelType.id || sensorId == evConnector.id) -> {
-                arrayOf("com.google.android.gms.permission.CAR_FUEL")
-            }
-            sensorId == odometerValue.id -> {
-                arrayOf("com.google.android.gms.permission.CAR_MILEAGE")
+        return carSensorsList.firstOrNull { it.sensor.id == sensorId }?.let {
+            if (isAutomotive) {
+                it.automotivePermissions.toTypedArray()
+            } else {
+                it.autoPermissions.toTypedArray()
             }
-            else -> emptyArray()
-        }
+        } ?: emptyArray()
     }
 
-    private lateinit var context: Context
+    fun isEnabled(context: Context, carSensor: CarSensor): Boolean {
+        this.context = context.applicationContext
 
-    private fun allDisabled(): Boolean = sensorsList.none { isEnabled(context, it) }
+        if (isAutomotive && !carSensor.automotiveEnabled || !isAutomotive && !carSensor.autoEnabled) {
```

Comment:
Do we need to add some parenthesis here so the conditions are properly evaluated?

Reply:
Adding them to improve readability